pytensor/link/numba/dispatch/subtensor.py

```diff
@@ -158,7 +158,7 @@ def advancedincsubtensor1_inplace(x, val, idxs):
             def advancedincsubtensor1_inplace(x, vals, idxs):
                 if not len(idxs) == len(vals):
                     raise ValueError("The number of indices and values must match.")
-                for idx, val in zip(idxs, vals):
+                for idx, val in zip(idxs, vals, strict=True):
```

Comment:
For all numba dispatch, we have to check if strict=True adds overhead in the Jitted code. If not sure, better to remove.